### ISSUE TITLE
Deal with whiteboard changes better when the Kanban block is removed

### ISSUE THREAD

**Issue by floatingatoll** [mozilla-it/bugzilla-kanbanize#9]
There are three similar scenarios we need to plan for:

* Someone clears a whiteboard status for a bug in a component we've configured the script to monitor via ```sync.cfg```.

If there's an open card linking to the bug, restore the whiteboard status link to the card. Otherwise, file a new card.

* Someone clears a whiteboard status for a bug in some other component, and Kanban is not cc'd.
* Someone clears a whiteboard status for a bug in some other component, and Kanban is cc'd.

This means that someone is trying to remove a bug from Kanban by altering Bugzilla fields. Let's honor that request. In both of these cases, we can detect that the bug had the Kanban link removed *after* we linked to the bug in the card, and treat that as an opportunity to clean up. In this scenario:

* Close the kanban card.
* Remove the Kanban CC, if present.

That should drastically simplify the process of de-assigning Kanban from bugs we send off to other queues, and remove all of the complications that we run into today with it recreating cards and all sorts of nonsense.

**Comment by floatingatoll** [comment]
As a side effect of this change, Kanban will stop filing duplicate cards when the whiteboard status is cleared, since it will find the card that links to the bug and reuse it.

**Comment by floatingatoll** [comment]
Pull #16 fixes the tendency to file duplicate cards rather than reusing existing cards.

**Comment by floatingatoll** [comment]
c72ec986e893ba3bb27e7ad7040054adf3741d28 adds the necessary framework for tracking where we learned about a bug, so that we can make the more intelligent decisions listed above.

**Comment by floatingatoll** [comment]
Pull #17 fixes an issue with source logging for Kanban.

**Comment by floatingatoll** [comment]
Merged #17, so that we now ignore blank whiteboard when a card references a bug that's been moved out of our watched components, is not cc'd to kanban in the bug, and has had the kanban link removed from the whiteboard.

**Comment by floatingatoll** [comment]
Merged #21, which implements some of the above logic. We now close cards that reference bugs  that left our components and aren't CC'd, because that means we can't track that bug anymore. If the bug returns to our components later, it will link the bug to the closed card. It ought to reopen the closed card, but that's not implemented yet.

```
[Wed Jun 08 05:41:19 2016] [WARN] Bug 1277639 already has a card 3072, updating whiteboard
[Wed Jun 08 05:41:20 2016] [WARN] [notimplemented] Should be reopening card 3072 and moving back to ready
[Wed Jun 08 05:41:20 2016] [INFO] [ 198/ 640] Card 3072 - Bug  1277639 - [search] kanbanize test bug ** [bug updated] **
```

**Comment by floatingatoll** [comment]
We use the cc: function extremely rarely, and should use it less so going forward. If work is ours, it's our bug. If it's not, the human being should remove the cc when clearing the whiteboard. Otherwise, kanban will reintroduce itself into the whiteboard (referencing the almost-certainly existing card that's still there).

Therefore, this is *technically* complete. The reopen functionality is a placeholder that emits the above warning, *but* this change results in that placeholder function being called - so this is functionally sufficient.
